### PR TITLE
HTTP.jl v0.8.0 compatibility

### PR DESCRIPTION
**Pull request by trigaut**
This PR resolves compatibility issues with HTTP.jl v0.8.0.
It removes REQUIRE and Manifest files.
I tagged two versions in my fork:
- v0.2.0: Version compatible with HTTP.jl v0.8.0
- v0.1.1: Version compatible with HTTP.jl [v0.6.0, v0.8.0)

### CODE REVIEW

File: src/Joseki.jl [hunk 1]
```diff
@@ -36,7 +36,11 @@ function stack(fns::Array{Function, 1}, endpoint::Function;
     end
     # Once we have this function we wrap it in a HandlerFunction which is used
     # to register routes
-    return HTTP.HandlerFunction(s)
+    return HTTP.Handlers.HandlerFunction(s)
+end
+
+function register!(router, method::String, url::String, handler)
+    eval(quote HTTP.@register($router, $method, $url, $handler) end)
```
Comment (amellnik) [inline src/Joseki.jl:43]: There was some discussion about removing the need for this in HTTP in a future release, but this looks good for now.

**Comment by amellnik** [comment]
This looks excellent -- thank you very much for your work on this.  I'm going to check a few additional things in the next day or so but I anticipate merging it with only minor changes (if any).

File: src/Joseki.jl (hunk 1 above)
Comment (trigaut) [reply src/Joseki.jl:43]: Good to know! And maybe there is already a more elegant way to deal with this?

**Comment by trigaut** [comment]
Thanks for the merge @amellnik . If you need any other help for the package I would be happy to contribute!

**Pull request merged**